### DISCUSSION TITLE
[Chapter 05] IdGenerator  method 안불러짐

### DISCUSSION THREAD

**Discussion by Youngju-Jang** [cowaine/spring-boot-for-msa#38] [Chapter 05. REST APIs with Spring MVC]
## 1. IdGenerator  method 
- 214 p 에서 `IdGenerator.creat()` 메소드를 통해 id 를 자동생성하는 코드가 보입니다.
- 뒷부분에서 따로 만든 클래스인것인지..? 인터페이스는 임포트가되는데 create 메소드가 불러지지 않아서 
임시적으로 `new Date().getTime();` 으로 사용하였습니다. 왜 안불러지는지 알 수 있을까요?

### 1. 잔디!
- 현재 레포지토리의 제 오리진브랜치에 커밋을 하면 일단 제 잔디에는 적용이 안되고,
- 현재 레포를 포크를떠서 제 레포에 따로 만들고, 싱크를 맞춰도 잔디적용이 안되고,
- 지난주에 풀리퀘스트 요청간것은 잔디적용이 되는것 같은데,
- 다른사람의 레포지토리에 같이 작업을 하는 경우 잔디채우기는 보통 어떤식으로 하나요?

### 2. domain ?
- `도메인`별로 설계를 한다~ 여기서 도메인이 게시글, 댓글, 알림, 이런식으로 서비스위주로 나누는 것이랑은 조금 다른개념이다~ 라는 말을 들은적이 있는데, `도메인`이라는 개념이 헷갈립니다.  
- 현재 저는 `domain` 폴더 밑에 `controller`, `dto`, 그리고 이번에  `ToDollarStringSerializer`클래스가 추가되면서 `utils` 폴더를 추가하였습니다. (차후 서비스별로 상위폴더 만들어 나눌예정)
  - 여기서 `domain` 이 ddd설계(도메인 주도 설계?) 할 때 그 domain이라고 생각하면 될까요?

**Comment by Irisation23** [comment]
``` 
1. 214 p 에서 IdGenerator.creat() 메소드를 통해 id 를 자동생성하는 코드가 보입니다.
```

- [김병부 님 코드 저장소](https://github.com/gilbutITbook/080264)을 확인 해 주세요.

**Reply by corock** [reply]
https://github.com/gilbutITbook/080264/blob/main/spring-boot-example/chapter05/src/main/java/com/springtour/example/chapter05/utils/IdGenerator.java
